### ISSUE TITLE
Add Cookie Consent tool to FreeBMD2 BETA

### ISSUE THREAD

**Issue by PatReynolds** [FreeUKGen/FreeBMD2#303]
Pat to set up (in part to test new cookie consent pathway).

a Cookie Consent is needed that does NOT allow essential cookies or functional cookies to be truned off and is a positive turn-off for Analytics  Advertising cookies.

**Comment by AlOneill** [comment]
@PatReynolds I happened to look at the "Cookie Policy" page today and noticed that it has a number of Accessibility issues — and at least one typo :-) See #390

It also looks like there could be work left to do on it: please do not forget to check for Accessibility (or ask me to do so) once the content is all in place. Thank you!

**Comment by PatReynolds** [comment]
Cannot be done (with Google consent generator) before the advert code is added, which cannot be done until password protection is removed.

**Comment by richpomfret** [comment]
@PatReynolds to look and see if there is Google test/dummy avert codes that can be used in order to set-up the cookie options correctly.

**Comment by PatReynolds** [comment]
Asked the community for help, but it looks as if we may need to take the password down while verification takes place.

**Comment by richpomfret** [comment]
Agreed in scrum to go ahead and remove the login to the site, therefore allowing us to verify the site and move ahead with implementing adverts.

**Comment by PatReynolds** [comment]
@Vino-S please let me know when you have removed the login.  Thanks, Pat

**Comment by richpomfret** [comment]
This has now moved forward - @PatReynolds to get the Adsense codes as next step.

**Comment by PatReynolds** [comment]
We no longer have to add the code - Google will serve the consents once I press the 'go' button, but this will be applied to _all_ FreeBMD sites, there is  no way to restrict it to Beta.

**Comment by richpomfret** [comment]
To test out first with FreeCEN.

**Comment by PatReynolds** [comment]
Cannot add the Google consent _just_ to Beta.

Looking at alternatives, I think we will need to pay.  Am I right in thinking 4 domains?  How many sessions total per month - do you know @Vino-S

**Comment by PatReynolds** [comment]
@PatReynolds to get sessions info from analtics and progress this story.

**Comment by PatReynoldsFUG** [comment]
I have set up one free trial - will emaily the script to you @Vino-S 

If you can let me know how many sessions per month for FreeBMD1, that would be great.

**Comment by PatReynoldsFUG** [comment]
![image.png](https://images.zenhubusercontent.com/63205da3a38f863bef8a7445/c572ddb0-800d-46f2-8918-b947b31e8c05)

**Comment by Vino-S** [comment]
@PatReynolds ,
We have atleast 420K sessions each month for FreeBMD1.

**Comment by PatReynoldsFUG** [comment]
@Vino-S we may have another option for Cookie consent - @DeniseColbert will tell you about it - so no need to implement this.

**Comment by KristinaGadzhieva** [comment]
Vino will contact Publift to ask for adding privacy feature

**Comment by KristinaGadzhieva** [comment]
@Vino-S could you please contact Publift?

**Comment by Vino-S** [comment]
Code is added and deployed to beta

**Comment by AlOneill** [comment]
I've commented on the Skiplinks issue that the link to "Privacy" (= Cookie Consent) has some issues. It breaks the expected functionality of the skiplink by getting focus on the first press of Tab. And because it is at the bottom of the screen, it is easy to miss. And should Privacy = Cookie Consent in the first place?

**Comment by KristinaGadzhieva** [comment]
@Vino-S @DeniseColbert  Checked in the meeting. We aren't seeing the cookie pop-up message on Beta
